include/algorithmFunctions.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+
+function calculatePoints($event){
+
+	$points = 0;
+
+	// these both get in the format of 'July 24, 2018'
+	$shortEventDate=formatDateToCalculate($event['dateOfEvent']);
+	$shortTodayDate= date("F j, Y");
+
+	//00:00 format
+	$fullEventDate = $event['dateOfEvent'];
+	$fullTodayDate = date('Y-m-d H:i:s');
+
+	$dateDiff= dateDifference($fullTodayDate, $event['dateOfEvent'])+1;
+
+
+	//NOTE: there is probably a better way to do this than a bunch of if statements,
+	// 		but just doing this for now.
+	if($shortEventDate==$shortTodayDate){
+		// echo "same day";
+		$points = $points +500;
+	}
+	else if($dateDiff == 1){ //aka tomorrow
+		$points = $points +450;
+	}
+	else if($dateDiff == 2){ //aka 2 days away
+		$points = $points +350;
+	}
+	else if($dateDiff == 3){
+		$points = $points +250;
+	}
+	else if($dateDiff == 4){
+		$points = $points +200;
+	}
+	else if($dateDiff == 5){
+		$points = $points +100;
+	}
+	else if(15>$dateDiff&&$dateDiff > 5){
+		$points = $points +100;
+	}
+	else if(60 > $dateDiff && $dateDiff >30){
+		$points = $points -300;
+	}
+	else if($dateDiff >60){
+		$points = $points -400;
+	}
+	else if($dateDiff <0){
+		$points = $points -500;
+	}
+
+	$usersGoing=getUsersGoingToThisEvent($event['eventID']);
+	$numPeopleGoing = count($usersGoing);
+
+	if($numPeopleGoing < 5){
+		$points = $points -100;
+	}
+	else if(10 > $numPeopleGoing && $numPeopleGoing > 5){
+		$points = $points +100;
+	}
+	else if(25 > $numPeopleGoing && $numPeopleGoing > 10){
+		$points = $points +150;
+	}
+	else if($numPeopleGoing > 25){
+		$points = $points +200;
+	}
+
+	$points=$points+($event['votes']*1.5);
+	$yMDH=date('Y-m-d-H');
+	updatePoints($event['eventID'], $points, $yMDH);
```

Comment:
This is a pretty simple algorithm, but sometimes simple is good. If you ever get anyone using this site, you'll probably want to keep an eye on the rankings and make adjustments based on your personal judgement (like, maybe you find that events tomorrow are just as important as events today, or that 5/10/25 aren't the right cut-offs for number of people going). But this seems like a great starting point.

Just to brainstorm ways this could be enhanced in the future:
- You could give extra weight to attendees that are more reputable (have more friends, have attended events in the past, etc.)
- You could personalize the algorithm to show each user different results depending on whether users they're friends with are going to the event. This would require storing a different score for each user
- You could let the hosts of events pay to boost their event's ranking to promote it

I realize you probably won't do any of those things, but it's fun to think about.